reconcile/openshift_tekton_resources.py

```diff
@@ -0,0 +1,353 @@
+import sys
+import logging
+import json
+from typing import Any, Iterable, Optional, Union
+
+import yaml
+import jinja2
+
+from reconcile import queries
+from reconcile import openshift_base as ob
+from reconcile.status import ExitCodes
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+from reconcile.utils.semver_helper import make_semver
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'openshift-tekton-resources'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+# it must be a single character due to resource max length
+OBJECTS_PREFIX = 'o'
+RESOURCE_MAX_LENGTH = 63
+
+# Defaults
+DEFAULT_DEPLOY_RESOURCES = {'requests': {'cpu': '50m',
+                                         'memory': '200Mi'},
+                            'limits': {'cpu': '200m',
+                                       'memory': '300Mi'}}
+SAAS_FILES_QUERY = """
+{
+  saas_files: saas_files_v2 {
+    path
+    name
+    configurableResources
+    pipelinesProvider {
+      name
+      provider
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
+      }
+    }
+  }
+}
+"""
+
+
+class OpenshiftTektonResourcesNameTooLongError(Exception):
+    pass
+
+
+class OpenshiftTektonResourcesBadConfigError(Exception):
+    pass
+
+
+# Builds a list of v2 saas files from qontract-server
+def get_saas_files(saas_file_name: Optional[str]) -> list[dict[str, Any]]:
+    saas_files = [
+        s for s in gql.get_api().query(SAAS_FILES_QUERY)['saas_files']
+        if s.get('configurableResources')]
+
+    if saas_file_name:
```

Comment:
Wouldn't it be easier to mark this field in the graphql schema with `isSearchable: true` and then query just for the saas_file_name? It'd be less code here and the integration would be much faster since you don't need to transmit and process a lot of data you'd be discarding.

Reply:
It's not worth the hassle. This is almost never going to be run like that, it will be run without options in prod. This is mainly a debug/develop option